### PR TITLE
[IMP] travis_install_nightly: Patch odoo to create unlogged DB tables

### LINKED ISSUE #279: Create UNLOGGED tables for faster creation
Use [SQL-CREATETABLE-UNLOGGED ](https://www.postgresql.org/docs/9.5/static/sql-createtable.html#SQL-CREATETABLE-UNLOGGED) recommended for [non-durability databases](https://www.postgresql.org/docs/9.5/static/non-durability.html)

Maybe patching odoo code

Or using
`ALTER TABLE "res_users" SET UNLOGGED;`
Note: The last command requires trace all parent tables in order to avoid a error similar to:
 - `ALTER TABLE "res_users" SET UNLOGGED;`
 - `ERROR:  could not change table "res_users" to unlogged because it references logged table "ir_model"`

UPDATED:
Running the following command
```bash
rgrep "CREATE TABLE" . --include=*.py --include=*.sql
./odoo/addons/base/base.sql:CREATE TABLE ir_actions (
./odoo/addons/base/base.sql:CREATE TABLE ir_act_window (primary key(id)) INHERITS (ir_actions);
./odoo/addons/base/base.sql:CREATE TABLE ir_act_report_xml (primary key(id)) INHERITS (ir_actions);
./odoo/addons/base/base.sql:CREATE TABLE ir_act_url (primary key(id)) INHERITS (ir_actions);
./odoo/addons/base/base.sql:CREATE TABLE ir_act_server (primary key(id)) INHERITS (ir_actions);
./odoo/addons/base/base.sql:CREATE TABLE ir_act_client (primary key(id)) INHERITS (ir_actions);
./odoo/addons/base/base.sql:CREATE TABLE res_users (
./odoo/addons/base/base.sql:CREATE TABLE res_groups (
./odoo/addons/base/base.sql:CREATE TABLE ir_module_category (
./odoo/addons/base/base.sql:CREATE TABLE ir_module_module (
./odoo/addons/base/base.sql:CREATE TABLE ir_module_module_dependency (
./odoo/addons/base/base.sql:CREATE TABLE ir_model_data (
./odoo/addons/base/base.sql:CREATE TABLE res_currency (
./odoo/addons/base/base.sql:CREATE TABLE res_company (
./odoo/addons/base/base.sql:CREATE TABLE res_partner (
./odoo/tools/sql.py:    cr.execute('CREATE TABLE "{}" (id SERIAL NOT NULL, PRIMARY KEY(id))'.format(tablename))
./odoo/fields.py:                CREATE TABLE "{rel}" ("{id1}" INTEGER NOT NULL,
./addons/base_import/static/csv/database_import_test.sql:CREATE TABLE companies (
./addons/base_import/static/csv/database_import_test.sql:CREATE TABLE persons (
```
There are 5 files where a database is created.

Apply a patch similar to:
 - `find . -type f -name "*.sql" -exec sed -i "s/CREATE TABLE/CREATE TABLE UNLOGGED/g" {} \;`
 - `find . -type f -name "*.py" -exec sed -i "s/CREATE TABLE/CREATE TABLE UNLOGGED/g" {} \;`

Note: Use the real name of files because of there is 1 sql and 2 py

### PR DESCRIPTION
**Pull request by moylop260**
This commit causes Odoo to be patched so all database tables are created
as unlogged, just after it's clonned.

It is applied just for runbot instance since that the records are
deleted after a docker stop if the postgreql service is started

This is compatible with the versions 7.0 to master (12.0).

Closes #279

### CODE REVIEW

File: travis/test_server.py [hunk 1]
```diff
@@ -355,6 +355,33 @@ def run_from_env_var(env_name_startswith, environ):
             raise RuntimeWarning("Return different to zero")
 
 
+def patch_odoo_unlogged_tables(server_path):
+    """ Patch Odoo to  make it to create all DB tables as unlogged """
```
Comment (moylop260) [inline travis/test_server.py]: Strip spaces

File: travis/test_server.py [hunk 2]
```diff
@@ -355,6 +355,33 @@ def run_from_env_var(env_name_startswith, environ):
...
+        'addons/base/data/base_data.sql',
+        'fields.py',
+        'models.py',
+        'tools/sql.py',
+        'osv/orm.py',
+    ]
+    if os.path.isdir(os.path.join(server_path, "odoo")):
+        odoo_subdir = "odoo"
+    else:
+        odoo_subdir = "openerp"
```
Comment (moylop260) [inline travis/test_server.py]: Dont use “else”

File: travis/test_server.py [hunk 3]
```diff
@@ -355,6 +355,33 @@ def run_from_env_var(env_name_startswith, environ):
...
+    for filename in filenames_to_patch:
+        filename = os.path.join(server_path, odoo_subdir, filename)
+        if not os.path.isfile(filename):
+            continue
+        with open(filename) as f_to_patch:
+            if not re.search("create table", f_to_patch.read(), re.IGNORECASE):
+                continue
+        print("\tPatchin file %s" % filename)
```
Comment (moylop260) [inline travis/test_server.py]: typo patching

File: travis/test_server.py [hunk 4]
```diff
@@ -405,6 +432,8 @@ def main(argv=None):
     odoo_full = os.environ.get("ODOO_REPO", "odoo/odoo")
     server_path = get_server_path(
         odoo_full, odoo_branch or odoo_version, travis_home)
+    if is_runbot:
```
Comment (moylop260) [inline travis/test_server.py]: if it is runbot and it is gitlab

File: travis/travis_after_tests_success [hunk 5]
```diff
@@ -84,6 +85,12 @@ def generate_pgbadger(exclude_regex=None):
 if os.environ.get('TESTS', '0') == '1' and \
         os.environ.get('TEST_ENABLE', '1') == '1' and not depends_error:
 
+    is_runbot = str2bool(os.environ.get('RUNBOT'))
+    if is_runbot:
+        # Stop PostgreSQL service. Even though the container is stopped once
+        # this script finishes, if changes are not saved before that, it
+        # could lead to inconsistencies on unlogged tables
+        subprocess.call(["/etc/init.d/postgresql", "stop"])
```
Comment (moylop260) [inline travis/travis_after_tests_success]: remove it
it is not needed anymore

**Comment by moylop260** [comment]
Create a dummy please

**Comment by moylop260** [comment]
Get last changes runbot-addons from oca to vx

**Comment by luisg123v** [comment]
@moylop260  
- I applied all requested changes
- Runbot changes are on https://github.com/Vauxoo/runbot-addons/pull/183
- There is a dummy, but Runbot changes need to be merged before it may be tested: https://git.vauxoo.com/vauxoo/enterprise-mx/merge_requests/355

**Comment by moylop260** [comment]
@luisg123v 
Could you create a PR to oca/mqt with this change please?

**Comment by moylop260** [comment]
For the record, We missed travis-ci (It is patching just for runbot, gitlab-ci but not for travis-ci)

**Pull request merged**